tests/unit/customizations/test_overridesslcommonname.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from awscli.testutils import mock, create_clidriver
+from awscli.customizations.overridesslcommonname import (
+    update_endpoint_url,
+    ssl_common_names,
+)
+
+import pytest
+
+
+test_cases = [
+    (service, region)
+    for service, regions in ssl_common_names.items()
+    for region in regions
+]
+
+
+@pytest.fixture
+def parsed_globals():
+    pg = mock.Mock()
```

Comment:
Instead of using `Mock` here, let's use `argparse.Namespace()`. The main concern in using `Mock()` without the spec argument is that it can swallow potential issues in the code as it happily accepts any property/method calls and this can lead to issues in updating the code down the road.

Reply:
yeah makes sense! i think i did this based off of other tests i saw in the code base, but agreed it would be good to use a more realistic that isn't magic and can accept anything as a property/attribute/method